pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -2012,30 +2016,32 @@ func TestReconcileTimeouts(t *testing.T) {
 		}}
 
 	for _, tc := range testcases {
-		d := test.Data{
-			TaskRuns: []*v1beta1.TaskRun{tc.taskRun},
-			Tasks:    []*v1beta1.Task{simpleTask},
-		}
-		testAssets, cancel := getTaskRunController(t, d)
-		defer cancel()
-		c := testAssets.Controller
-		clients := testAssets.Clients
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{
```

Comment:
we need similar refactoring here 😿, the ones we did in PR #2876 and may be share some of the common functions 🤔

Reply:
When I work on separating common function like we discussed in PR #2876, I will try and see if its possible to refactor `taskrun_test.go`